tika-parsers/src/main/java/org/apache/tika/parser/hwp/HwpV5Parser.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.tika.parser.hwp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Set;
+
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.mime.MediaType;
+import org.apache.tika.parser.AbstractParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.sax.XHTMLContentHandler;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+
+public class HwpV5Parser extends AbstractParser {
+
+	private static final long serialVersionUID = 1L;
+	
+	private static final Set<MediaType> SUPPORTED_TYPES = Collections.singleton(MediaType.application("x-hwp-v5"));
+    public static final String HWP_MIME_TYPE = "application/x-hwp-v5";
+    
+    private HwpTextExtractorV5 extractor;
+    
+    public HwpV5Parser() {
+    	extractor = new HwpTextExtractorV5();
```

Comment:
Parsers need to be thread-safe.  HwpTextExtractorV5 stores a poifs as an instance variable.  So, you need to create a new HwpTextExtractorV5 for each parse, or remove the poifs instance variable.

Reply:
removed the poifs instance variable